src/api/XRInputSource.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XRInputPose from './XRInputPose';
+import { mat4_identity, mat4_fromRotationTranslation, mat4_copy } from '../math';
+
+const PRIVATE = Symbol('@@webxr-polyfill/XRInputSource');
+
+const HEAD_CONTROLLER_RIGHT_OFFSET = new Float32Array([0.155, -0.465, -0.35]);
+const HEAD_CONTROLLER_LEFT_OFFSET = new Float32Array([-0.155, -0.465, -0.35]);
+
+export class XRInputSourceImpl {
```

Comment:
I think this class should live within the WebVRDevice.js file as it's polyfill-implementation-specific (or used by WebVRDevice, living at `./src/devices/GamepadInput.js` or something), but could be convinced otherwise